server/tls/index.js

```diff
@@ -0,0 +1,56 @@
+const grpc = require('grpc');
+const { readFileSync } = require('fs');
+
+function getCredentials() {
+  if (process.env.TEMPORAL_TLS_KEY_PATH === undefined) {
+    console.log('establishing insecure connection...');
+    return { credentials: grpc.credentials.createInsecure(), options: {} };
+  }
+
+  console.log('establishing secure connection using TLS...');
+
+  let credentials;
+  if (process.env.TEMPORAL_TLS_CERT_PATH === undefined) {
+    throw Error('TLS certificate is not provided');
+  }
+
+  const pk = readFileSync(process.env.TEMPORAL_TLS_KEY_PATH);
+  const cert = readFileSync(process.env.TEMPORAL_TLS_CERT_PATH);
+
+  let ca;
+  if (process.env.TEMPORAL_TLS_CA_PATH) {
+    ca = process.env.TEMPORAL_TLS_CA_PATH;
+    caContent = readFileSync(ca);
+  }
+
+  const verifyHost = [true, 'true'].includes(
+    process.env.TEMPORAL_TLS_ENABLE_HOST_VERIFICATION
```

Comment:
instead of having this, why not just check whether `TEMPORAL_TLS_SERVER_NAME` is defined?